ow_regolith/src/RegolithSpawner.cpp

```diff
@@ -310,11 +315,19 @@ void RegolithSpawner::onLinkStatesMsg(const LinkStates::ConstPtr &msg)
   }
 
   auto pose_it = begin(msg->pose) + distance(begin(msg->name), name_it);
-  tf::quaternionMsgToTF(pose_it->orientation, m_scoop_orientation);
+  quaternionMsgToTF(pose_it->orientation, m_scoop_orientation);
 }
 
 void RegolithSpawner::onModDiffVisualMsg(const modified_terrain_diff::ConstPtr& msg)
 {
+  // check if visual terrain modification was caused by the scoop
+  Vector3 scoop_bottom(quatRotate(m_scoop_orientation, m_scoop_down));
+  if (tfDot(scoop_bottom, Vector3(0.0, 0.0, -1.0)) < 0.0)
```

Comment:
Why not save and use a normalized gravity vector that you queried for during initialization?
```suggestion
  if (tfDot(scoop_bottom, gravity_norm) < 0.0)
```
(instead of making the assumption that z is down)

Reply:
I actually did it this way at first, but I realized assuming that the scoop bottom has to be aligned with the gravity vector is equally as restricting as assuming it must align with `-z`, so I used the less complicated method. Furthermore, since `ow_dynamic_terrain` uses essentially the same method to decide if collision terrain deformation should occur, this approach is more consistent with approaches used in other parts of our code. This will also have to be revisited at some point anyways when/if we implement sloped scooping.